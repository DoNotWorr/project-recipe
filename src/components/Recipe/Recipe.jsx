import React from "react";
import { useParams } from "react-router-dom";
import styles from "./Recipe.module.css";

export default function Recipe(props) {
  const { id } = useParams();

  const recipe = props.getRecipeById(id);

  return (
    <div className={styles.recipe}>
      <h1>{recipe.name}</h1>
      <img src={recipe.images.big} alt={recipe.name} />
      <h2>Ingredients</h2>
      <ul className={styles.ingredients}>
        {recipe.ingredients.map((ingredient) => (
          <li>
            <p>
              {ingredient.name} {ingredient.quantity} {ingredient.unit}
            </p>
          </li>
        ))}
      </ul>

      <h2>Instructions</h2>
      <ol className={styles.instructions}>
        {recipe.instructions.map((instruction) => (
          <li>{instruction}</li>
        ))}
      </ol>
    </div>
  );
}
