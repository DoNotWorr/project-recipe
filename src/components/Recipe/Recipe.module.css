.ingredients {
  display: flex;
  flex-direction: column;
  list-style: disc inside;
  margin: 0;
}
.ingredients ul li {
  text-align: left;
}

.instructions {
  display: flex;
  list-style: decimal inside;
  flex-direction: column;
  padding: 0 0 6vh 0;
}

.recipe h1 {
  padding: 4vh;
}
