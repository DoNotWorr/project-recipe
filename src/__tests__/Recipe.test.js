import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Recipe from "../components/Recipe/Recipe";
import { BrowserRouter } from "react-router-dom";

const sampleRecipe = {
  id: "1",
  name: "Avokado Sandwich",
  ingredients: [
    {
      name: "flour",
      quantity: 2,
      unit: "table spoon",
    },
    {
      name: "sugar",
      quantity: 10,
      unit: "gram",
    },
    {
      name: "milk",
      quantity: 1,
      unit: "deciliter",
    },
  ],
  instructions: [
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec blandit eu quam quis porta.",
    "Duis magna elit, sodales ut nisl et, efficitur tristique lacus.",
  ],
  category: "Appetizers",
  images: {
    small: "/images/avokado_small.jpg",
    big: "/images/avokado_big.jpg",
  },
};

const mockFunction = (id) => sampleRecipe;

describe("<Recipe />", () => {
  test("should work", () => {
    render(
      <BrowserRouter>
        <Recipe getRecipeById={mockFunction} key={sampleRecipe.id} />
      </BrowserRouter>
    );

    const h2 = screen.getByText(sampleRecipe.name);
    const img = screen.getByAltText(sampleRecipe.name);

    expect(h2).toBeInTheDocument();
    expect(img).toBeInTheDocument();
  });
});
