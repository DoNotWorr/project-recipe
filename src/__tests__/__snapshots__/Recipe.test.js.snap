// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Recipe /> should work 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="recipe"
      >
        <div
          class="imgContainer"
        >
          <h2>
            Avokado Sandwich
          </h2>
          <img
            alt="Avokado Sandwich"
            class="image"
            src="/images/avokado_big.jpg"
          />
        </div>
        <div
          class="ingrContainer"
        >
          <h3>
            Ingredients
          </h3>
          <ul
            class="ingredients"
          >
            <li>
              flour
               
              2
               
              table spoon
            </li>
            <li>
              sugar
               
              10
               
              gram
            </li>
            <li>
              milk
               
              1
               
              deciliter
            </li>
          </ul>
        </div>
        <div
          class="instrContainer"
        >
          <h3>
            Instructions
          </h3>
          <ol
            class="instructions"
          >
            <li>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec blandit eu quam quis porta.
            </li>
            <li>
              Duis magna elit, sodales ut nisl et, efficitur tristique lacus.
            </li>
          </ol>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="recipe"
    >
      <div
        class="imgContainer"
      >
        <h2>
          Avokado Sandwich
        </h2>
        <img
          alt="Avokado Sandwich"
          class="image"
          src="/images/avokado_big.jpg"
        />
      </div>
      <div
        class="ingrContainer"
      >
        <h3>
          Ingredients
        </h3>
        <ul
          class="ingredients"
        >
          <li>
            flour
             
            2
             
            table spoon
          </li>
          <li>
            sugar
             
            10
             
            gram
          </li>
          <li>
            milk
             
            1
             
            deciliter
          </li>
        </ul>
      </div>
      <div
        class="instrContainer"
      >
        <h3>
          Instructions
        </h3>
        <ol
          class="instructions"
        >
          <li>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec blandit eu quam quis porta.
          </li>
          <li>
            Duis magna elit, sodales ut nisl et, efficitur tristique lacus.
          </li>
        </ol>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
